src/main.c

```diff
@@ -2425,9 +2434,44 @@ void handle_movement(double dt) {
         if (glfwGetKey(g->window, GLFW_KEY_RIGHT)) s->rx += m;
         if (glfwGetKey(g->window, GLFW_KEY_UP)) s->ry += m;
         if (glfwGetKey(g->window, GLFW_KEY_DOWN)) s->ry -= m;
+        
+        if (autoWalkState == GLFW_PRESS)
+        {
+        	if(*is_auto_walking_ptr == false && *allow_next_autowalk_press_ptr == true) 
+        	{
+        	
+        	*is_auto_walking_ptr = true;
+        	*do_auto_walk_ptr = true;
+        	}
+        	else if(*is_auto_walking_ptr == true && *allow_next_autowalk_press_ptr == true) 
+        	{
+        	
+        	*is_auto_walking_ptr = false;
+        	*do_auto_walk_ptr = false;
+        	
+        	}
+		*allow_next_autowalk_press_ptr = false;
+        	
+        	//*allow_next_autowalk_press_ptr = false; //saftey net, takes care of holding down button
+        	
+        }
+        else if (autoWalkState == GLFW_RELEASE) {
+        	*allow_next_autowalk_press_ptr = true;
+        }//end auto walk
+        	
     }
     float vx, vy, vz;
-    get_motion_vector(g->flying, sz, sx, s->rx, s->ry, &vx, &vy, &vz);
+    
+    //New system overrides this code piece
```

Comment:
Having a reason for commenting out code is alright. However according to Uncle Bob, you should really not do this in order to keep the code clean. If this note is really important, I suggest you document it elsewhere & not in the program itself. Same goes for all the other commented out code in the program. I do see on line 2455 as of now there is also commented out code of which does not have a reason & I suggest just removing that then if it has no importance.

Reply:
Done. Thanks for the feedback!